api/swagger.yml

```diff
@@ -1632,8 +1632,47 @@ components:
       required:
         - installation_id
         - reports
-
-
+    ExternalPrincipalList: 
+      type: object
+      required:
+        - pagination
+        - results
+      properties:
+        pagination:
+          $ref: "#/components/schemas/Pagination"
+        results:
+          type: array
+          items:
+            $ref: "#/components/schemas/ExternalPrincipal"
+    ExternalPrincipalSettings:
+      type: object
+      additionalProperties:
+        type: string
+      description: Additional settings to be consumed by the remote authenticator
+    ExternalPrincipalCreation:
+      type: object
+      properties:
+        settings:
+          type: object
+          items:
+            $ref: "#/components/schemas/ExternalPrincipalSettings"
+    ExternalPrincipal:
+      type: object
+      required:
+        - user_id
+        - id
+      properties:
+        id:
+          type: string
+          description: A unique identifier for the external principal
+        user_id:
+          type: string
+          description: |
+            lakeFS user ID to associate with an external principal.
```

Comment:
Email? something else?

Reply:
I have no idea what's the question?

Reply:
I think we have many id types for a user in Treeverse, right? I may be wrong..